candle-binding/src/classifiers/lora/intent_lora.rs

```diff
@@ -0,0 +1,138 @@
+//! Intent classification with LoRA adapters
+//!
+//! High-performance intent classification using real model inference
+
+use crate::core::{processing_errors, ModelErrorType, UnifiedError};
+use crate::model_architectures::lora::bert_lora::HighPerformanceBertClassifier;
+use crate::model_error;
+use candle_core::Result;
+use std::time::Instant;
+
+/// Intent classifier with real model inference (merged LoRA models)
+pub struct IntentLoRAClassifier {
+    /// High-performance BERT classifier for intent classification
+    bert_classifier: HighPerformanceBertClassifier,
+    /// Confidence threshold for predictions
+    confidence_threshold: f32,
+    /// Intent labels mapping
+    intent_labels: Vec<String>,
+    /// Model path for reference
+    model_path: String,
+}
+
+/// Intent classification result
+#[derive(Debug, Clone)]
+pub struct IntentResult {
+    pub intent: String,
+    pub confidence: f32,
+    pub processing_time_ms: u64,
+}
+
+impl IntentLoRAClassifier {
+    /// Create new intent classifier using real model inference
+    pub fn new(model_path: &str, use_cpu: bool) -> Result<Self> {
+        // Load labels from model config
+        let intent_labels = Self::load_labels_from_config(model_path)?;
+        let num_classes = intent_labels.len();
+
+        // Load the high-performance BERT classifier for merged LoRA models
+        let classifier = HighPerformanceBertClassifier::new(model_path, num_classes, use_cpu)
+            .map_err(|e| {
+                let unified_err = model_error!(
+                    ModelErrorType::LoRA,
+                    "intent classifier creation",
+                    format!("Failed to create BERT classifier: {}", e),
+                    model_path
+                );
+                candle_core::Error::from(unified_err)
+            })?;
+
+        Ok(Self {
+            bert_classifier: classifier,
+            confidence_threshold: 0.7,
```

Comment:
would the threshold come from config? https://github.com/vllm-project/semantic-router/blob/main/config/config.yaml#L3